python/ql/src/experimental/Security/CWE-090/examples/example_good2.py

```diff
@@ -0,0 +1,17 @@
+from flask import request, Flask
+import ldap3
+from ldap3.utils.dn import escape_rdn
+from ldap3.utils.conv import escape_filter_chars
+
+
+@app.route("/normal")
+def normal():
+    unsafe_dn = "dc=%s" % request.args['dc']
+    unsafe_filter = "(user=%s)" % request.args['username']
+
+    safe_dn = escape_rdn(unsafe_dn)
+    safe_filter = escape_filter_chars(unsafe_filter)
```

Comment:
If you apply escaping to the _entire_ DN, that will also escape the `=`, so you should do something like this instead (I'm also using `.format` instead of %-formatting, since that's so much more common in Python code these days)

```suggestion
    safe_dn = "dc={}".format(escape_rdn(request.args['dc']))
    safe_filter = "(user={})".format(escape_filter_chars(request.args['username']))
```

Reply:
Spot on! I focused on the actual taint flow and didn't take into account that those DNs/Filters would throw syntax errors. However instead of declaring the variable in the same line, I do think it'd be clearer and faster to understand if there's an order like the following:

```py
unsafe_dc = request.args['dc']
safe_dc = escape_rdn(unsafe_dc)
dn = "dc={}".format(safe_dc)
# ...
conn.search(dn, ...)
```

Since the __only__ change between (good|bad) tests will be the appearance of `safe_dc = escape_rdn(unsafe_dc)`. 

Does it make sense?

Reply:
Yes that makes a lot of sense, and I think that does probably look better than what I suggested :+1: